src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -345,6 +345,86 @@ object MatrixIR {
       keepF(mv, keep)
     })
   }
+
+  def collectColsByKey(typ: MatrixType): (MatrixType, MatrixValue => MatrixValue) = {
+    val newColValueType = TStruct(typ.colValueStruct.fields.map(f => f.copy(typ = TArray(f.typ, required = true))))
```

Comment:
Why is `required = True`? Since the documentation doesn't say otherwise, I'd assumed a missing value becomes a missing array element.

Reply:
It's possible I still don't completely understand missingness. My intention here is that a field `foo: T` becomes a field `foo: Array[T]`, where the array can't be missing, but the missingness of the entries of the array is the same as the missingness of the original field.